### ISSUE TITLE
Some changes in release 17.0.0 were mistakenly left out

### ISSUE THREAD

**Issue by acknosyn** [primer/components#757]
Some changes in release [17.0.0](https://github.com/primer/components/releases/tag/v17.0.0) were mistakenly left out. 🕵️‍♂️

I went down a rabbit hole trying to figure it all out. Hopefully my explanation makes sense.

**TL;DR: some changes in the v17.0.0 release are missing, like the Heading breaking change. See missing changes below.**

## How I found out
I was updating versions and was looking at the breaking changes to the `Heading` component in release **17.0.0**. The changes made `Heading` components render as an `h2` tag by default instead of `h1`.

After updating, I couldn't find the change it talked about. I thought I was going mad because `Headings` were still rendering as an `h1` without an `as` prop telling it to.

After some digging, I eventually confirmed the changes were not there and found out why.

## Explanation
Work included in the **v17.0.0** release, like the `Heading` component updates, merged into the `major` branch were reverted by a later merge from an out-of-sync component branch; the new `Pagination` component. (I'll refer to the `mkt/pagination-component` branch as the `pagination` branch.)

The `pagination` branch started from a commit after **v16.3.0**, from `master`. The `major` branch received a merge from `master` to update it to that commit too. So with the first commit on `pagination`, it was up-to-date with `major`, which is good.

A bunch of work is done on `pagination`. Then after some time, `major` starts getting updated with new commits. `pagination` is being worked on in parallel with different commits. So by the end of the work being done on `pagination`, it was out-of-date with `major`. `major` had a bunch of commits which `pagination` didn't have. 

So before the work on `pagination` was done and merged into `major`, it received an update from `major`. The merge should've brought `pagination` up-to-date with the new `major` commits, but it didn't. 

There would've been a lot of merge conflicts with files that had changed in `major`. For some reason, that merge (`ba41cd4`) was empty, not sure why. Maybe the merge conflicts were dismissed?

`pagination` was still out-of-date with the `major` changes. Then the commit (`95acd04`) to merge `pagination` into `major` reverted a bunch of changes. Because of this, all commits in the `major` branch starting from `b8b43cf` to this `pagination` branch merge, were reverted.

It took me a long time to fully understand everything, so let me know if something doesn't make sense.

Here's an image showing the branches in GitKraken.
![Screen Shot 2020-04-14 at 8](https://user-images.githubusercontent.com/8602276/79202467-c6dbbf00-7e8d-11ea-8eac-87517dd46f4b.png)

## Missing changes
In the [diff](https://github.com/primer/components/compare/v16.3.0...v17.0.0) between the previous release (**v16.3.0**) and **v17.0.0**, you can see the changes made in `major` are in the history. But they were reverted by the mentioned `pagination` branch merge (`95acd04`).

All commits on `major` starting from `b8b43cf` to `fbd493f` were reverted.

These are the commits that were not included in the **v17.0.0** release:
- [b8b43cf](https://github.com/primer/components/commit/b8b43cf33174dd9afc30fbbf7d8431e899c0190e) (Merge pull request #732 from primer/update-box-shadow)
- [be6ed30](https://github.com/primer/components/commit/be6ed30a9c9697c4bed6c59a23cabcb85a010837) (Merge pull request #736 from primer/mkt/dialog-header-extraction)
- [98af424](https://github.com/primer/components/commit/98af424a209c73bab963b75623e80a23cefc8cee) (Merge branch 'taylorcjohnson-heading-update' of https://github.com/ta…)
- [4446796](https://github.com/primer/components/commit/4446796a648e9ff410cb67b508aa80600eeb3972) (Update test snapshots)
- [fbd493f](https://github.com/primer/components/commit/fbd493fef532cf4d245aaabf7371834e98fa33a7) (Fix lint)

## Fix
As the fix, I've cherry picked those missing commits and dealt with the merge conflicts from newer changes to the files (like removing theme from test snapshots, `559af65`).

I'll submit a PR which attempts to fix this. **Edit: PR at #758**

Cheers,
Acknosyn

(P.S. Sorry if this comes across as blamey—not my intention, just being clear with where things happen.)

**Comment by acknosyn** [comment]
Fixed with the v18.0.0 release, cheers.
